### PR TITLE
search/filter component

### LINKED ISSUE #25: SelectionOptions
### SelectionOptions
Get an array of the options to generate the Select options.

- props: array of values

- [ ]  It will be used in Pages:

- Sign in.
- Signup.
- Profile.
- Search.

- [ ]  Ant Design UI link : 
[Select Component](https://ant.design/components/select/)

- [ ] ** UI Image From Design**

image

### LINKED ISSUE #92: Search component
- [ ] 2 selection filters
- [ ] search button
- [ ] Page Head Speech

### PR DESCRIPTION
**Pull request by zeinjendeya**
- select component for search
- filter component with two filters and a search button
- data file for services' types and locations

### CODE REVIEW

**Review by Ahmed-aed — DISMISSED** [review]

File: client/src/Components/Select/index.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+const { Option } = Select;
+
+const SelectComponent = ({ options, handleChange, textDefault }) => (
+  <Select
+    size="large"
+    defaultValue="select a service type"
+    onChange={handleChange}
+  >
+    <Option value={textDefault}>{textDefault}</Option>
```
Comment (mariamisa) [inline client/src/Components/Select/index.js]: delete this line, it's not the right way to set the default value
[antd-select](https://ant.design/components/select/#header)

File: client/src/Components/Search/index.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  const [location, setLocation] = useState('null');
+
+  handleSearch(service, location);
+
+  const handleService = (value) => {
+    setService(value);
+  };
+  const handleLocation = (value) => {
+    setLocation(value);
+  };
```
Comment (mariamisa) [inline client/src/Components/Search/index.js]: * This component is the presentational component that doesn't contain state, it just uses the function from props to set these input value from the parent component -- remove all this code
* you can use setService and SetLocation functions directly, why you put them inside a function?
```
// props
const Filter = ({ handleSearch, setService , setLocation }) //these functions pas from parent component
```

File: client/src/Components/Search/index.js [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    <div className="filter-container">
+      <Select
+        className="filter-select"
+        textDefault={serviceTypes[0]}
+        handleChange={handleService}
+        options={serviceTypes}
+      />
+      <p className="split-text">in</p>
+      <Select
+        className="filter-select"
```
Comment (mariamisa) [inline client/src/Components/Search/index.js]: you can set the className as default on a select component

File: client/src/Components/Search/index.js [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+import PropTypes from 'prop-types';
+import { SearchOutlined } from '@ant-design/icons';
+import Button from '../Button';
+import Select from '../Select';
+
+import './style.css';
+
+import { locations, serviceTypes } from '../../Utils/data';
+
+const Filter = ({ handleSearch }) => {
```
Comment (mariamisa) [inline client/src/Components/Search/index.js]: rename the component name to SearchBar

File: client/src/Components/Search/index.js [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  const [service, setService] = useState('null');
```
Comment (mariamisa) [inline client/src/Components/Search/index.js]: if you want to set a null value to any state keep it empty ``` useState(); ``` or ``` useState(null) ``` in this way you set a string, not a null value

**Review by mariamisa — CHANGES_REQUESTED** [review]

File: client/src/Components/Search/index.js (hunk 3 above)
Comment (zeinjendeya) [reply client/src/Components/Search/index.js]: I don't think I understand what you mean but these classes are for a already set style.

**Review by mariamisa — APPROVED** [review]

**Review by mariamisa — APPROVED** [review]

**Pull request merged**